lib/bazel/commands/target_list.go

```diff
@@ -0,0 +1,191 @@
+	targets := maps.Keys(t.targets)
+	sort.Strings(targets)
+	return targets
+}
+
+func (t *targetStatus) PrettyPrint(w io.Writer) error {
+	running := t.Filter("no_status")
+	passed := t.Filter("passed")
+	flaky := t.Filter("flaky")
+	timedOut := t.Filter("timeout")
+	failed := t.Filter("failed")
+	incomplete := t.Filter("incomplete")
+	infraFailure := t.Filter("remote_failure")
+	buildFailure := t.Filter("failed_to_build")
+	toolHalted := t.Filter("tool_halted_before_testing")
+
+	// When printing, print the more important failures towards the bottom, which
+	// are more visible to the user.
+
+	if t.finished {
+		// Successes are the least interesting when the invocation has finished
+		if len(passed.targets) > 0 {
+			prettyPrintTargetList(w, passed, "Completed targets:", "🟢")
+		}
+		if len(incomplete.targets) > 0 {
+			prettyPrintTargetList(w, incomplete, "Incomplete/aborted targets:", "❓")
+		}
+		if len(flaky.targets) > 0 {
+			prettyPrintTargetList(w, flaky, "Flaky targets:", "🪫")
+		}
+		if len(toolHalted.targets) > 0 {
+			prettyPrintTargetList(w, toolHalted, "Halted before testing:", "🛑")
+		}
+		if len(failed.targets) > 0 {
+			prettyPrintTargetList(w, failed, "Failed targets:", "❌")
+		}
+		if len(timedOut.targets) > 0 {
+			prettyPrintTargetList(w, timedOut, "Timed out targets:", "⏰")
+		}
+		if len(infraFailure.targets) > 0 {
+			prettyPrintTargetList(w, infraFailure, "Infra failures:", "🚨")
+		}
+		if len(buildFailure.targets) > 0 {
+			prettyPrintTargetList(w, buildFailure, "Broken targets:", "💣")
+		}
+	} else {
+		// Running targets are the least interesting when the invocation has
+		// not yet finished
+		if len(running.targets) > 0 {
+			prettyPrintTargetList(w, running, "Running targets:", "🟡")
+		}
+		if len(passed.targets) > 0 {
+			prettyPrintTargetList(w, passed, "Completed targets:", "🟢")
+		}
+		if len(incomplete.targets) > 0 {
+			prettyPrintTargetList(w, incomplete, "Incomplete targets:", "❓")
+		}
+		if len(flaky.targets) > 0 {
+			prettyPrintTargetList(w, flaky, "Flaky targets:", "🪫")
+		}
+		if len(toolHalted.targets) > 0 {
+			prettyPrintTargetList(w, toolHalted, "Halted before testing:", "🛑")
+		}
+		if len(failed.targets) > 0 {
+			prettyPrintTargetList(w, failed, "Failed targets:", "❌")
+		}
+		if len(timedOut.targets) > 0 {
+			prettyPrintTargetList(w, timedOut, "Timed out targets:", "⏰")
+		}
+		if len(infraFailure.targets) > 0 {
+			prettyPrintTargetList(w, infraFailure, "Infra failures:", "🚨")
+		}
+		if len(buildFailure.targets) > 0 {
+			prettyPrintTargetList(w, buildFailure, "Broken targets:", "💣")
+		}
+	}
+
+	return nil
+}
+
+func prettyPrintTargetList(w io.Writer, ts *targetStatus, header string, icon string) {
+	io.WriteString(w, header+"\n")
+	for _, t := range ts.Sorted() {
+		io.WriteString(w, icon+" "+t+"\n")
+	}
+	io.WriteString(w, "\n")
+}
+
+func (t *targetStatus) Print(w io.Writer) error {
+	for _, name := range t.Sorted() {
+		io.WriteString(w, name+"\n")
```

Comment:
nit: `fmt.Fprintln` would eliminate the need to concat the newline

Reply:
Changed all the `io.WriteString`s to `fmt.Fprintln` or `fmt.Fprintf`